components/scream/src/physics/common/physics_saturation_impl.hpp

```diff
@@ -10,13 +10,76 @@ namespace physics {
  * Implementation of saturation functions. Clients should NOT #include
  * this file, #include physics_functions.hpp instead.
  */
+
+template <typename S, typename D>
+KOKKOS_FUNCTION
+void  Functions<S,D>
+::check_temp(const Spack& temp, const std::string& fname)
+{
+  Smask is_neg_temp = (temp <= 0);
+  if (is_neg_temp.any()){
+    scream::error::runtime_abort("Error: Called from:"+ fname +"; Temperature has <= 0 values");
+  }
+  /*
+  Smask is_nan_temp = isnan(temp);
+  if (is_nan_temp.any()){
+    scream::error::runtime_abort("Error: Called from:"+ fname +"; Temperature has NaN values");
+  }
+
+  Smask is_finite_temp = isfinite(temp);
+  if (is_finite_temp.any()){
+      scream::error::runtime_abort("Error: Called from:"+ fname +";  Temperature has infinite values");
+    }*/
+}
+
+template <typename S, typename D>
+KOKKOS_FUNCTION
+typename Functions<S,D>::Spack
+Functions<S,D>::MurphyKoop_svp(const Spack& t, const bool ice)
+{
+  //Formuas used below are from the following paper:
+  //Murphy, D. M., and T. Koop (2005), Review of the vapour pressures of ice
+  //and supercooled water for atmospheric applications, Quart J. Roy. Meteor.
+  //Soc., 131(608), 1539–1565,
+
+  Spack result;
+  const auto tmelt = C::Tmelt;
+  Smask ice_mask = (t < tmelt) && ice;
+  Smask liq_mask = !ice_mask;
+
+  if (ice_mask.any()) {
+    //Equation (7) of the paper
+    // (good down to 110 K)
+    //creating array for storing coefficients of ice sat equation
+    const Scalar ic[]= {9.550426, 5723.265, 3.53068, 0.00728332}; // coefficients
+    Spack ice_result = exp(ic[0] - (ic[1] / t) + (ic[2] * log(t)) - (ic[3] * t));
+
+    result.set(ice_mask, ice_result);
+  }
+
+  if (liq_mask.any()) {
+    //Equation (10) of the paper
+    // (good for 123 < T < 332 K)
+    //creating array for storing coefficients of liq sat equation
+    const Scalar lq[] = {54.842763, 6763.22, 4.210, 0.000367, 0.0415, 218.8, 53.878, //coefficients
+			 1331.22, 9.44523, 0.014025 };
```

Comment:
maybe it's just github formatting ,but it looks weird for "//coefficients" comment to be stuffed halfway through the array assignment rather than on the line above with the rest of the comment it refers to?

Reply:
I will remove this comment since the comment above the line is quite clear.